### PR TITLE
Admin Refresh

### PR DESCRIPTION
**Pull request by tn3rb**
This is some work I've been picking at for awhile to update the EE admin UI to better match with the changes in EDTR.

It's mostly cosmetic, but there are a few structural/layout tweaks as well, but those have been done in a backwards compatible manner as can be seen by looking at https://barista.eventespresso.com/ which has most EE plugins running. 

ping me in chat if you need access to that site

### CODE REVIEW

File: admin_pages/events/Events_Admin_List_Table.class.php [hunk 1]
```diff
@@ -42,21 +44,22 @@ protected function _set_properties()
             'ajax'     => true, // for now
             'screen'   => $this->_admin_page->get_current_screen()->id,
         );
+        $approved_egistrations = esc_html__('Approved Registrations', 'event_espresso');
         $this->_columns = array(
-            'cb'              => '<input type="checkbox" />', // <-- this checkbox is actually replaced by WP core
+            'cb'              => '<input type="checkbox" />',
+            // 'status'          => '',
```
Comment (knazart) [inline admin_pages/events/Events_Admin_List_Table.class.php:50]:
```suggestion
```

File: admin_pages/events/Events_Admin_List_Table.class.php [hunk 2]
```diff
@@ -127,43 +125,35 @@ protected function _get_row_class($item)
...
      */
-    public function column_id(EE_Event $item)
+    public function column_id(EE_Event $event)
     {
-        $content = $item->ID();
-        $content .= '  <span class="show-on-mobile-view-only">' . $item->name() . '</span>';
-        return $content;
+        $content = $event->ID();
+        $content .= '<span class="show-on-mobile-view-only">';
+        $content .= $this->column_name($event, false);;
```
Comment (knazart) [inline admin_pages/events/Events_Admin_List_Table.class.php:154]:
```suggestion
        $content .= $this->column_name($event, false);
```

File: admin_pages/events/Events_Admin_List_Table.class.php [hunk 3]
```diff
@@ -396,128 +389,140 @@ function (EE_Term $category) {
...
      * @throws InvalidArgumentException
      * @throws InvalidDataTypeException
      * @throws InvalidInterfaceException
+     * @throws ReflectionException
      */
-    public function column_tkts_sold(EE_Event $item)
+    public function column_tkts_sold(EE_Event $event)
     {
-        return EEM_Ticket::instance()->sum(array(array('Datetime.EVT_ID' => $item->ID())), 'TKT_sold');
+        $content = EEM_Ticket::instance()->sum(array(array('Datetime.EVT_ID' => $event->ID())), 'TKT_sold');
```
Comment (knazart) [inline admin_pages/events/Events_Admin_List_Table.class.php:486]:
```suggestion
        $content = EEM_Ticket::instance()->sum([['Datetime.EVT_ID' => $event->ID()]], 'TKT_sold');
```

File: admin_pages/registrations/EE_Registrations_List_Table.class.php [hunk 4]
```diff
@@ -422,119 +408,250 @@ protected function _total_registrations_today()
...
-    public function column__REG_ID(EE_Registration $item)
+    public function column_ATT_fname(EE_Registration $registration, $prep_content = true)
     {
-        $attendee = $item->attendee();
-        $content  = $item->ID();
-        $content  .= '<div class="show-on-mobile-view-only">';
-        $content  .= '<br>';
-        $content  .= $attendee instanceof EE_Attendee
+        $attendee      = $registration->attendee();
+        $edit_link_url  = EE_Admin_Page::add_query_args_and_nonce(
```
Comment (knazart) [inline admin_pages/registrations/EE_Registrations_List_Table.class.php:472]:
```suggestion
        $attendee       = $registration->attendee();
        $edit_link_url  = EE_Admin_Page::add_query_args_and_nonce(
```

File: admin_pages/registrations/EE_Registrations_List_Table.class.php [hunk 5]
```diff
@@ -422,119 +408,250 @@ protected function _total_registrations_today()
...
+                'espresso_registrations_trash_registrations',
+                $registration->ID()
+            )
+        ) {
+            $action['action'] = 'trash_registrations';
+            $trash_link_url    = EE_Admin_Page::add_query_args_and_nonce(
+                $action,
+                REG_ADMIN_URL
+            );
+            $actions['trash'] = '<a class="ee-aria-tooltip" href="'
```
Comment (knazart) [inline admin_pages/registrations/EE_Registrations_List_Table.class.php:547]:
```suggestion
            $action['action'] = 'trash_registrations';
            $trash_link_url   = EE_Admin_Page::add_query_args_and_nonce(
                $action,
                REG_ADMIN_URL
            );
            $actions['trash'] = '<a class="ee-aria-tooltip" href="'
```

File: admin_pages/registrations/EE_Registrations_List_Table.class.php [hunk 6]
```diff
@@ -422,119 +408,250 @@ protected function _total_registrations_today()
...
+            // restore registration link
+            if (
+                EE_Registry::instance()->CAP->current_user_can(
+                    'ee_delete_registration',
+                    'espresso_registrations_restore_registrations',
+                    $registration->ID()
+                )
+            ) {
+                $action['action']   = 'restore_registrations';
+                $restore_link_url    = EE_Admin_Page::add_query_args_and_nonce(
```
Comment (knazart) [inline admin_pages/registrations/EE_Registrations_List_Table.class.php:562]:
```suggestion
                $action['action']   = 'restore_registrations';
                $restore_link_url   = EE_Admin_Page::add_query_args_and_nonce(
```

File: admin_pages/registrations/EE_Registrations_List_Table.class.php [hunk 7]
```diff
@@ -422,119 +408,250 @@ protected function _total_registrations_today()
...
+            }
+            if (
+                EE_Registry::instance()->CAP->current_user_can(
+                    'ee_delete_registration',
+                    'espresso_registrations_ee_delete_registrations',
+                    $registration->ID()
+                )
+            ) {
+                $action['action']  = 'delete_registrations';
+                $delete_link_url    = EE_Admin_Page::add_query_args_and_nonce(
```
Comment (knazart) [inline admin_pages/registrations/EE_Registrations_List_Table.class.php:580]:
```suggestion
                $delete_link_url   = EE_Admin_Page::add_query_args_and_nonce(
```

File: admin_pages/registrations/EE_Registrations_List_Table.class.php [hunk 8]
```diff
@@ -422,119 +408,250 @@ protected function _total_registrations_today()
...
-            : $item->get_i18n_datetime('REG_date');
-        $view_link    .= '<br><span class="ee-status-text-small">'
-                         . EEH_Template::pretty_status($this->_transaction_details['status'], false, 'sentence')
-                         . '</span>';
-        return $view_link;
+            : $registration->get_i18n_datetime('REG_date');
+
+        // $content .= ' <span class="ee-status-text-small">'
+        //             . EEH_Template::pretty_status($this->_transaction_details['status'], false, 'sentence')
+        //             . '</span>';
```
Comment (knazart) [inline admin_pages/registrations/EE_Registrations_List_Table.class.php:634]:
```suggestion
```

File: admin_pages/registrations/assets/espresso_registrations_admin.css [hunk 9]
```diff
@@ -311,178 +407,72 @@
...
-    height : 200px;
+    height: 200px;
 }
 
 .ee-batch-message-send-form .shortcodes-info-container {
-    padding          : 5px 20px 5px 20px;;
-    margin-bottom    : 10px;
-    background-color : #F5F5F5;
-    border-radius    : 4px;
+    padding: 5px 20px 5px 20px;;
```
Comment (knazart) [inline admin_pages/registrations/assets/espresso_registrations_admin.css:469]:
```suggestion
    padding: 5px 20px 5px 20px;
```

File: caffeinated/admin/extend/registrations/EE_Event_Registrations_List_Table.class.php [hunk 10]
```diff
@@ -264,148 +269,161 @@ protected function _add_view_counts()
...
-        $name_link .= '</span>';
-        $actions = array();
-        $DTT_ID = $this->_cur_dtt_id;
-        $latest_related_datetime = empty($DTT_ID) && ! empty($this->_req_data['event_id']) && $item instanceof EE_Registration
-            ? $item->get_latest_related_datetime()
-            : null;
-        $DTT_ID = $latest_related_datetime instanceof EE_Datetime
+        // $name_link               .= '<br><span class="ee-status-text-small">';
+        // $name_link               .= EEH_Template::pretty_status($registration->status_ID(), false, 'sentence');
+        // $name_link               .= '</span>';
```
Comment (knazart) [inline caffeinated/admin/extend/registrations/EE_Event_Registrations_List_Table.class.php:419]:
```suggestion
```

**Review by knazart — CHANGES_REQUESTED** [review]
Found a couple but minor syntax errors and had a few things pointed out. 

Otherwise looks 👍

File: admin_pages/transactions/Transactions_Admin_Page.core.php [hunk 11]
```diff
@@ -982,17 +980,18 @@ public function getActionButtons(EE_Transaction $transaction = null)
...
             apply_filters('FHEE__Transactions_Admin_Page__getActionButtons__actions', $actions, $transaction)
         );
         if ($actions) {
-            $content = '<ul>';
-            $content .= '<li>' . implode('</li><li>', $actions) . '</li>';
-            $content .= '</uL>';
+            // $content = '<ul>';
+            // $content .= '<li>' . implode('</li><li>', $actions) . '</li>';
+            // $content .= '</uL>';
+            $content .= implode('', $actions);
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply admin_pages/transactions/Transactions_Admin_Page.core.php:994]: nope totally unnecessary with modern css.  we used to have to add extra markup in order to achieve specific layouts, but now with things like flexbox and grid it's totally unnecessary.

File: admin_pages/messages/Messages_Template_List_Table.class.php [hunk 12]
```diff
@@ -326,7 +337,7 @@ protected function _get_context_links(EE_Message_Template_Group $item)
             return '';
         }
         // we want to display the contexts in here, so we need to set them up
-        $c_label           = $item->context_label();
+        // $c_label           = $item->context_label();
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply admin_pages/messages/Messages_Template_List_Table.class.php:340]: oh i think it just looked like it was removed in the diff, but that's still used on line 362

**Comment by garthkoyle** [comment]
* [ ] Can the Add New Event button be blue instead of green? I don’t think we use a green action button elsewhere (or at least I haven’t noticed).
* [ ] https://github.com/eventespresso/event-espresso-core/issues/3742
* [ ] The Transactions > `Apply Discount Code` button could use some love.
* [ ] In Messages > Default Templates or Custom Templates, can we make the Recipients column wider, and narrow down the Description column of the Messages Templates view? 99.999% of the time we're looking to click the correct message context, not read the description.
* [ ] The Messages Templates editor Update Actions button style needs some love:
* [ ] The Venue “Update Venue” `Move to Trash` button needs some love
* [ ] The Promotions `Save and Close` button needs some love
* [ ] The “People Admin” `Add Person` form needs some love with the Last Name:
https://www.screencast.com/t/cXM4TVPK
* [ ] The Save Person Record `Move to Trash` button needs some love

File: admin_pages/messages/Messages_Template_List_Table.class.php (hunk 12 above)
[the remark this answers is not in the document]
Comment (tn3rb) [reply admin_pages/messages/Messages_Template_List_Table.class.php:340]: oops... read your comment wrong... ya this can be (and has been) removed

**Pull request closed without merging**